genrl/deep/agents/dqn/dqn.py

```diff
@@ -341,18 +349,34 @@ def get_td_loss(self) -> torch.Tensor:
             q_target_s_a_prime = q_target_next_state_values.gather(
                 1, action_next.unsqueeze(1)
             ).squeeze(1)
-            expected_q_value = reward + self.gamma * q_target_s_a_prime * (1 - done)
+            expected_q_value = reward + self.gamma * q_target_s_a_prime.reshape(-1, 1) * (1 - done)
 
         else:
             q_values = self.model(state)
-            q_value = q_values.gather(1, action.unsqueeze(1)).squeeze(1)
+            # print(action)
+            q_value = q_values.gather(1, action).squeeze(1)
 
             q_next_state_values = self.target_model(next_state)
             q_s_a_prime = q_next_state_values.max(1)[0]
-            expected_q_value = reward + self.gamma * q_s_a_prime * (1 - done)
-
+            # print(reward.shape, q_s_a_prime.shape, done.shape)
+            expected_q_value = reward + self.gamma * q_s_a_prime.reshape(-1,1) * (1 - done)
+
+#Commenting this conflict, as I'm not sure what is to be done here.
+#<<<<<<< vecenvs
+#        if self.prioritized_replay and (not self.categorical_dqn):
+#            loss = (q_value - expected_q_value.detach()).pow(2) * weights
+#            priorities = loss + 1e-5
+#             loss = loss.mean()
+#             self.replay_buffer.update_priorities(indices, priorities.data.cpu().numpy())
+
+#         elif (not self.prioritized_replay) and (not self.categorical_dqn):
+#             loss = (q_value.unsqueeze(1) - expected_q_value.detach()).pow(2).mean()
+#         # loss = F.smooth_l1_loss(q_value,expected_q_value)
+# 
+# =======
         if self.categorical_dqn:
             loss = -(Variable(projection_dist) * dist.log()).sum(1).mean()
+# >>>>>>> master
         else:
             if self.prioritized_replay:
                 loss = (q_value - expected_q_value.detach()).pow(2) * weights
```

Comment:
@Het-Shah Can you check this? I wasnt sure how to resolve the conflict.

(Confused between the many if statements)

Reply:
Yea sure.